codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java

```diff
@@ -157,8 +160,7 @@ protected String getBaseDirectory() {
 
     @Override
     protected String getDiskFilename() {
-        File file = new File(filename);
-        return file.getName();
+        return String.valueOf(fileNameCounter.incrementAndGet());
```

Comment:
I have to agree there, `File.createTempFile` is much better, using `AtomicLong` won't guarantee name uniqueness, also another advantage of `File.createTempFile` is that it can mark the file for deletion at JVM exit as a fail-safe mechanism.